Add meaningful CSS class to AutosizeInput wrapper div

trumbitta opened the pull request:
This little div:

```html
<div
  class="select__value-container select__value-container--has-value css-g1d714-ValueContainer"
>
  <div class="select__single-value css-1uccc91-singleValue">Ocean</div>

  <!-- This one right here below -->
  <div class="css-b8ldur-Input">

    <div class="select__input" style="display: inline-block">
      <input [...] />
      [...]
    </div>
  </div>
</div>
```

bears some CSS rules that would be much easier to customize (via class name / styled-components) if it had a proper, meaningful, fixed class name.

This little PR adds that class name:

```html
<div
  class="select__value-container select__value-container--has-value css-g1d714-ValueContainer"
>
  <div class="select__single-value css-1uccc91-singleValue">Ocean</div>

  <!-- This one right here below -->
  <div class="css-b8ldur-Input select__input-wrapper">

    <div class="select__input" style="display: inline-block">
      <input [...] />
      [...]
    </div>
  </div>
</div>
```

Without it, I'm having to do this:

```css
    /*
      This is the parent of the div that hides the real HTML input element.
      The problem is that this element causes unwanted margin, height, padding but
      it has no proper name / class while its direct child has a proper name / class but at the time
      of writing we can't target a parent by its child in CSS (https://caniuse.com/css-has)

      So, I guess I'll just 🤞
    */
    div:nth-child(2) {
      margin: 0;
      padding: 0;
    }
```

but I would rather not :D

JedWatson commented:
Hey @trumbitta thanks for this PR!

We're currently working through the updates we're making for 5.0 (see #4559) and have two thoughts on this:

1. We are looking into replacing `react-input-autosize` with a css-only solution now that we have dropped IE11 support, which would clean this whole part of the component up considerably. If we do that, we **may** be able to remove one of these two wrapping divs entirely 
2. If we do keep both wrapping divs, I'd be very happy to merge this PR, but for consistency with other wrapper class names we should change it to `input-container` and not `input-wrapper`

Will update you shortly with how this is going

trumbitta commented:
Hey @JedWatson thanks for the heads-up!

While you decide about the best course of action, I thought I'd better help by doing what I could and  so I updated the name to `input-container` and rebased the branch to get the latest updates from master in.

ebonow commented:
@trumbitta just wanted to say thanks for this PR. 

One of the goals moving into a new major version was to identify any breaking changes and autosizeinput was one of those long overdue changes. Much of the Input component was thus overhauled as a part of this PR. https://github.com/JedWatson/react-select/pull/4625

The impact of the referenced PR is that it duplicates this functionality.

Some other things to note:
- The DOM structure was simplified so the Input component is only one div wrapping an input
- inputClassName was added as a prop to allow more control in adding styles to the input
- `.prefix__input-container` refers to the div wrapper whereas `.prefix__input` will now refer to the input element. This will likely cause some initial confusion, but feel it's much more consistent with the styles.

That all said, I agree with the intention of this PR, but will close this as this functionality will is included as a part of another PR.

trumbitta commented:
Hey, thanks for the heads-up!

I'm excited for the upcoming release, but at the same time I don't know when I will be able to use it since a sizable slice of our users unfortunately still uses IE11.

Is there any chance you'll release a patch version with this PR before going forward with the next major? 🤞🤞🤞
